src/Compilers/CSharp/Portable/Lowering/ClosureConversion/ClosureConversion.cs

```diff
@@ -465,6 +465,7 @@ private void SynthesizeClosureMethods()
                     originalMethod,
                     nestedFunction.BlockSyntax,
                     lambdaId);
+                synthesizedMethod.AfterCreate(Diagnostics);
```

Comment:
>synthesizedMethod.AfterCreate(Diagnostics); [](http://example.com/codeflow?start=16&length=43)

Is this method going to bring something new? I.e. Can we figure this out from the lambda/local function and do that where other similar checks are performed (see usage of LocalRewriter.CheckRefReadOnlySymbols )?